<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot coverage in nuclear and cytoplasmic RNA for candidate probe sequences. — plot_coverage • brainflowprobes</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Plot coverage in nuclear and cytoplasmic RNA for candidate probe sequences. — plot_coverage" />
<meta property="og:description" content="plot_coverage plots the coverage of nuclear and cytoplasmic RNA from
adult and prenatal human prefrontal cortex across a candidate or series of
candidate probe sequences for BrainFlow. If the sequence spans splice
junctions, the plot will include the introns. A good candidate sequence will
be highly and evenly expressed in nuclear RNA." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">brainflowprobes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/brainflowprobes-vignette.html">brainflowprobes user guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LieberInstitute/brainflowprobes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot coverage in nuclear and cytoplasmic RNA for candidate probe sequences.</h1>
    <small class="dont-index">Source: <a href='https://github.com/LieberInstitute/brainflowprobes/blob/master/R/plot_coverage.R'><code>R/plot_coverage.R</code></a></small>
    <div class="hidden name"><code>plot_coverage.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>plot_coverage</code> plots the coverage of nuclear and cytoplasmic RNA from
adult and prenatal human prefrontal cortex across a candidate or series of
candidate probe sequences for BrainFlow. If the sequence spans splice
junctions, the plot will include the introns. A good candidate sequence will
be highly and evenly expressed in nuclear RNA.</p>
    </div>

    <pre class="usage"><span class='fu'>plot_coverage</span><span class='op'>(</span>
  <span class='va'>REGION</span>,
  PDF <span class='op'>=</span> <span class='st'>"regionCoverage_fractionedData.pdf"</span>,
  OUTDIR <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>,
  COVERAGE <span class='op'>=</span> <span class='cn'>NULL</span>,
  CODING_ONLY <span class='op'>=</span> <span class='cn'>FALSE</span>,
  VERBOSE <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>REGION</th>
      <td><p>Either a single hg19 genomic sequence including the chromosome,
start, end, and optionally strand separated by colons (e.g.,
'chr20:10199446-10288068:+'), or a string of sequences. Must be character.
Chromosome must be proceeded by 'chr'.</p></td>
    </tr>
    <tr>
      <th>PDF</th>
      <td><p>The name of the PDF file. Defaults to
<code>regionCoverage_fractionedData.pdf</code>.</p></td>
    </tr>
    <tr>
      <th>OUTDIR</th>
      <td><p>The default directory where <code>PDF</code> will be saved to.</p></td>
    </tr>
    <tr>
      <th>COVERAGE</th>
      <td><p>The output of <a href='brainflowprobes_cov.html'>brainflowprobes_cov</a> for the input
<code>REGION</code>. Defaults to <code>NULL</code> but it can be pre-computed and saved
separately since this is the step that takes the longest to run. Also, this
is the only step that depends on rtracklayer's functionality for reading
BigWig files which does not run on Windows OS. So it could be run on a
non-Windows machine, saved, and then shared with Windows users.</p></td>
    </tr>
    <tr>
      <th>CODING_ONLY</th>
      <td><p>A logical vector of length 1 specifying whether to
subset the Annotated Genes to only the coding genes. That is, whether to
subset the genes by whether they have a non-NA <code>CSS</code> value. The Annotated
Genes are downloaded with <code><a href='https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html'>GenomicState::GenomicStateHub()</a></code>.</p></td>
    </tr>
    <tr>
      <th>VERBOSE</th>
      <td><p>A logical value indicating whether to print updates from the
process of loading the data from the BigWig files.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>plot_coverage</code> plots all input sequences using
<code><a href='https://rdrr.io/pkg/derfinderPlot/man/plotRegionCoverage.html'>derfinderPlot::plotRegionCoverage()</a></code>. It returns a plot for each
input candidate sequence listed in REGION. Each plot includes coverage of
the sequence(s) in nuclear (N) and cytoplasmic (C) RNA isolated from adult
(A) and fetal (F) prefrontal cortex, sequenced using two different library
preparation methods. PolyA+ libraries (P) were generated using selection
for polyadenylated transcripts, and RiboZero (R) libraries were generated
using a ribosomal RNA depletion protocol.</p>
<p>Each plot also shows the overlapping genes beneath the coverage, and the
genomic location. The title lists the nearest gene, the position of the
sequence relative to the gene's canonical transcriptional start site (TSS),
and further annotation information as described in the 'region' column from
<code><a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>bumphunter::matchGenes()</a></code>.</p>
<p><code>plot_coverage</code> saves the results as regionCoverage_fractionedData.pdf
in a temporary directory unless otherwise specified with <code>OUTDIR</code>.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Amanda J Price</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'></span>
<span class='r-in'><span class='co'>## Here we use the pre-saved example coverage data such that this example</span></span>
<span class='r-in'><span class='co'>## will run fast!</span></span>
<span class='r-in'><span class='fu'>plot_coverage</span><span class='op'>(</span><span class='st'>"chr20:10286777-10288069:+"</span>,</span>
<span class='r-in'>    COVERAGE <span class='op'>=</span> <span class='va'>four_panels_example_cov</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:31:49 annotateRegions: counting</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:31:51 annotateRegions: annotating</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:31:51 Completed! Check for /tmp/RtmpftNADa/regionCoverage_fractionedData.pdf.</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Without using COVERAGE, this function reads BigWig files from the web</span></span>
<span class='r-in'><span class='co'>## using rtracklayer and this functionality is not supported on Windows</span></span>
<span class='r-in'><span class='co'>## machines.</span></span>
<span class='r-in'><span class='kw'>if</span> <span class='op'>(</span><span class='va'>.Platform</span><span class='op'>$</span><span class='va'>OS.type</span> <span class='op'>!=</span> <span class='st'>"windows"</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'>    <span class='fu'>plot_coverage</span><span class='op'>(</span><span class='st'>"chr20:10286777-10288069:+"</span>,</span>
<span class='r-in'>        PDF <span class='op'>=</span> <span class='st'>"regionCoverage_fractionedData_fromScratch.pdf"</span></span>
<span class='r-in'>    <span class='op'>)</span></span>
<span class='r-in'><span class='op'>}</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:13 annotateRegions: counting</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:13 annotateRegions: annotating</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:13 getRegionCoverage : attempting to load coverage data from 'files'.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:13 fullCoverage: processing chromosome chr20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:14 loadCoverage: finding chromosome lengths</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:14 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br1113C1_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:15 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br1113N1_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:16 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br2046C_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:17 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br2046N_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:18 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br2074C_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:19 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br2074N_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:20 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5339C1_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:21 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5339N1_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:22 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5340C1_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:23 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5340N1_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:24 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5341C1_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:25 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5341N1_polyA.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:26 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br1113C1_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:27 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br2046C_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:28 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br2046N_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:29 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br2074C_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:30 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br2074N_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:31 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5339C1_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:32 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5339N1_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:34 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5340C1_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:35 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5340N1_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:36 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5341C1_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:37 loadCoverage: loading BigWig file http://brain-flow-rna.s3.us-east-2.amazonaws.com/Sep/Br5341N1_RiboZero.bw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:38 loadCoverage: applying the cutoff to the merged data</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:38 filterData: originally there were 63025520 rows, now there are 63025520 rows. Meaning that 0 percent was filtered.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:38 getRegionCoverage: processing chr20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:38 getRegionCoverage: done processing chr20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-11-23 04:32:38 Completed! Check for /tmp/RtmpftNADa/regionCoverage_fractionedData_fromScratch.pdf.</span>
<span class='r-in'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## These examples will take a few minutes to run!</span></span>
<span class='r-in'><span class='fu'>plot_coverage</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span></span>
<span class='r-in'>    <span class='st'>"chr20:10286777-10288069:+"</span>,</span>
<span class='r-in'>    <span class='st'>"chr18:74690788-74692427:-"</span>,</span>
<span class='r-in'>    <span class='st'>"chr19:49932861-49933829:-"</span></span>
<span class='r-in'><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>candidates</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span></span>
<span class='r-in'>    <span class='st'>"chr20:10286777-10288069:+"</span>,</span>
<span class='r-in'>    <span class='st'>"chr18:74690788-74692427:-"</span>,</span>
<span class='r-in'>    <span class='st'>"chr19:49932861-49933829:-"</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## General syntax:</span></span>
<span class='r-in'><span class='fu'>plot_coverage</span><span class='op'>(</span><span class='va'>candidates</span>,</span>
<span class='r-in'>    PDF <span class='op'>=</span> <span class='st'>"PDF_file.pdf"</span>,</span>
<span class='r-in'>    OUTDIR <span class='op'>=</span> <span class='st'>"/path/to/directory/"</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='fu'>plot_coverage</span><span class='op'>(</span><span class='st'>"chr20:10286777-10288069:+"</span>,</span>
<span class='r-in'>    PDF <span class='op'>=</span> <span class='st'>"PDF_file.pdf"</span>, OUTDIR <span class='op'>=</span> <span class='st'>"/path/to/directory/"</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Explore the effect of changing CODING_ONLY</span></span>
<span class='r-in'><span class='co'>## Check how gene name and distance to TSS changes in the title of the plot</span></span>
<span class='r-in'><span class='co'>## (everything else stays the same)</span></span>
<span class='r-in'><span class='va'>cov</span> <span class='op'>&lt;-</span> <span class='fu'><a href='brainflowprobes_cov.html'>brainflowprobes_cov</a></span><span class='op'>(</span><span class='st'>"chr10:135379301-135379311:+"</span><span class='op'>)</span></span>
<span class='r-in'><span class='fu'>plot_coverage</span><span class='op'>(</span><span class='st'>"chr10:135379301-135379311:+"</span>, COVERAGE <span class='op'>=</span> <span class='va'>cov</span><span class='op'>)</span></span>
<span class='r-in'><span class='fu'>plot_coverage</span><span class='op'>(</span><span class='st'>"chr10:135379301-135379311:+"</span>,</span>
<span class='r-in'>    COVERAGE <span class='op'>=</span> <span class='va'>cov</span>,</span>
<span class='r-in'>    PDF <span class='op'>=</span> <span class='st'>"coding_only_plot_coverage"</span>, CODING_ONLY <span class='op'>=</span> <span class='cn'>TRUE</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'><span class='op'>}</span></span>
<span class='r-in'></span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Amanda Price.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


