<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Print relevant info about candidate probe sequence. — region_info • brainflowprobes</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Print relevant info about candidate probe sequence. — region_info" />
<meta property="og:description" content="region_info returns annotation of a single potential probe sequence or
list of sequences and, if specified, prints the resuts in a .csv file." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">brainflowprobes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/brainflowprobes-vignette.html">brainflowprobes user guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LieberInstitute/brainflowprobes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Print relevant info about candidate probe sequence.</h1>
    <small class="dont-index">Source: <a href='https://github.com/LieberInstitute/brainflowprobes/blob/master/R/region_info.R'><code>R/region_info.R</code></a></small>
    <div class="hidden name"><code>region_info.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>region_info</code> returns annotation of a single potential probe sequence or
list of sequences and, if specified, prints the resuts in a .csv file.</p>
    </div>

    <pre class="usage"><span class='fu'>region_info</span><span class='op'>(</span>
  <span class='va'>REGION</span>,
  CSV <span class='op'>=</span> <span class='cn'>TRUE</span>,
  SEQ <span class='op'>=</span> <span class='cn'>TRUE</span>,
  OUTDIR <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>,
  CODING_ONLY <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>REGION</th>
      <td><p>Either a single hg19 genomic sequence including the chromosome,
start, end, and optionally strand separated by colons (e.g.,
<code>'chr20:10199446-10288068:+'</code>), or a string of sequences to be annotated.
Must be character. Chromosome must be proceeded by 'chr'.</p></td>
    </tr>
    <tr>
      <th>CSV</th>
      <td><p>A <code>logical(1)</code> value indicating if the results should be exported
in a .csv file.</p></td>
    </tr>
    <tr>
      <th>SEQ</th>
      <td><p>A `logical(1)`` value indicating if the base sequence should be
returned.</p></td>
    </tr>
    <tr>
      <th>OUTDIR</th>
      <td><p>If a .csv file is to be exported, this parameter indicates the
path where the file should be saved. By default the file will be
saved in a temporary directory.</p></td>
    </tr>
    <tr>
      <th>CODING_ONLY</th>
      <td><p>A logical vector of length 1 specifying whether to
subset the Annotated Genes to only the coding genes. That is, whether to
subset the genes by whether they have a non-NA <code>CSS</code> value. The Annotated
Genes are downloaded with <code><a href='https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html'>GenomicState::GenomicStateHub()</a></code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>This function annotates all input sequences using
<code><a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>bumphunter::matchGenes()</a></code>. It returns a data frame where each
row is a genomic sequence specified in REGION. The columns
c('seqnames', 'start', 'end', 'width', 'strand') list the chromosome,
range, sequence length, and strand of the REGION. The columns c('name',
'annotation', 'description', 'region', 'distance', 'subregion',
'insideDistance', 'exonnumber', 'nexons', 'UTR', 'geneL', 'codingL',
'Geneid', 'subjectHits') are described in
<code><a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>bumphunter::matchGenes()</a></code> documentation.</p>
<p>If SEQ=TRUE, a column 'Sequence' will be included. This is recommended for
sending the probe sequence to be synthesized.</p>
<p>If CSV=TRUE, a .csv file called region_info.csv will be saved to a
temporary directory unless otherwise specified in <code>OUTDIR</code>.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Amanda J Price</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'>region_info</span><span class='op'>(</span><span class='st'>"chr20:10286777-10288069:+"</span>, CSV <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Completed! If CSV=TRUE, check for region_info.csv in the temporary</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> directory (i.e. tempdir()) unless otherwise specified in OUTDIR.</span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqnames    start      end width strand   name</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1    chr20 10286777 10288069  1293      + SNAP25</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                                                                                                                                                                                                                                                                                                                                                        annotation</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1 NM_001322902 NM_001322903 NM_001322904 NM_001322905 NM_001322906 NM_001322907 NM_001322908 NM_001322909 NM_001322910 NM_003081 NM_130811 NP_001309831 NP_001309832 NP_001309833 NP_001309834 NP_001309835 NP_001309836 NP_001309837 NP_001309838 NP_001309839 NP_003072 NP_570824 XM_005260808 XM_017028021 XM_017028022 XP_005260865 XP_016883510 XP_016883511</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   description      region distance                subregion insideDistance</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1 overlaps 3' overlaps 3'    87299 overlaps exon downstream              0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   exonnumber nexons            UTR geneL codingL          Geneid</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1         10     10 overlaps 3'UTR 88588   30705 ENSG00000132639</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Sequence</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1 GCTGATTCCAACAAAACCAGAATTGATGAGGCCAACCAACGTGCAACAAAGATGCTGGGAAGTGGTTAAGTGTGCCCACCCGTGTTCTCCTCCAAATGCTGTCGGGCAAGATAGCTCCTTCATGCTTTTCTCATGGTATTATCTAGTAGGTCTGCACACATAACACACATCAGTCCACCCCCATTGTGAATGTTGTCCTGTGTCATCTGTCAGCTTCCCAACAATACTTTGTGTCTTTTGTTCTCTCTTGGTCTCTTTCTTTCCAAAGGTTGTACATAGTGGTCATTTGGTGGCTCTAACTCCTTGAGGTCTTGAGTTTCATTTTTCATTTTCTCTCCTCGGTGGCATTTGCTGAATAACAACAATTTAGGAATGCTCAATGTGCTGTTGATTCTTTCAATCCACAGTATTGTTCTTGTAAAACTGTGACATTCCACAGAGTTACTGCCACGGTCCTTTGAGTGTCAGGCTCTGAATCTCTCAAAATGTGCCGTCTTTGGTTCCTCATGGCTGTTATCTGTCTTTATGATTTCATGATTAGACAATGTGGAATTACATAACAGGCATTGCACTAAAAGTGATGTGATTTATGCATTTATGCATGAGAACTAAATAGATTTTTAGATTCCTACTTAAACAAAAACTTTCCATGACAGTAGCATACTGATGAGACAACACACACACACACAAAACAACAGCAACAACAACAGAACAACAACAAAGCATGCTCAGTATTGAGACACTGTCAAGATTAAGTTATACCAGCAAAAGTGCAGTAGTGTCACTTTTTTCCTGTCAATATATAGAGACTTCTAAATCATAATCATCCTTTTTTAAAAAAAAGAATTTTAAAAAAGATGGATTTGACACACTCACCATTTAATCATTTCCAGCAAAATATATGTTTGGCTGAAATTATGTCAAATGGATGTAATATAGGGTTTGTTTGCTGCTTTTGATGGCTATGTTTTGGAGAGAGCAATCTTGCTGTGAAACAGTGTGGATGTAAATTTTATAAGGCTGACTCTTACTAACCACCATTTCCCCTGTGGTTTGTTATCAGTACAATTCTTTGTTGCTTAATCTAGAGCTATGCACACCAAATTGCTGAGATGTTTAGTAGCTGATAAAGAAACCTTTTAAAAAAATAATATAAATGAATGAAATATAAACTGTGAGATAAATATCATTATAGCATGTAATATTAAATTCCTCCTGTCTCCTCTGTCAGTTTGTGAAGTGATTGACATTTTGTAGCTAGTTTAAAATTATTAAAAATTATAGACTCCAGAT</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## You can easily transform this data.frame to a GRanges object</span></span>
<span class='r-in'><span class='fu'>GenomicRanges</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 1 range and 14 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       seqnames            ranges strand |        name             annotation</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;            &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]    chr20 10286777-10288069      + |      SNAP25 NM_001322902 NM_0013..</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       description      region  distance                subregion insideDistance</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>          &lt;factor&gt;    &lt;factor&gt; &lt;numeric&gt;                 &lt;factor&gt;      &lt;numeric&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1] overlaps 3' overlaps 3'     87299 overlaps exon downstream              0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>       exonnumber    nexons            UTR     geneL   codingL          Geneid</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>        &lt;numeric&gt; &lt;integer&gt;       &lt;factor&gt; &lt;numeric&gt; &lt;numeric&gt;     &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1]         10        10 overlaps 3'UTR     88588     30705 ENSG00000132639</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                     Sequence</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                  &lt;character&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   [1] GCTGATTCCAACAAAACCAG..</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'>region_info</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span></span>
<span class='r-in'>    <span class='st'>"chr20:10286777-10288069:+"</span>,</span>
<span class='r-in'>    <span class='st'>"chr18:74690788-74692427:-"</span>,</span>
<span class='r-in'>    <span class='st'>"chr19:49932861-49933829:-"</span></span>
<span class='r-in'><span class='op'>)</span>,</span>
<span class='r-in'>CSV <span class='op'>=</span> <span class='cn'>FALSE</span>, SEQ <span class='op'>=</span> <span class='cn'>FALSE</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Completed! If CSV=TRUE, check for region_info.csv in the temporary</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> directory (i.e. tempdir()) unless otherwise specified in OUTDIR.</span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqnames    start      end width strand    name</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1    chr20 10286777 10288069  1293      +  SNAP25</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2    chr18 74690788 74692427  1640      -     MBP</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3    chr19 49932861 49933829   969      - SLC17A7</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                        annotation</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1                                 NM_001322902 NM_001322903 NM_001322904 NM_001322905 NM_001322906 NM_001322907 NM_001322908 NM_001322909 NM_001322910 NM_003081 NM_130811 NP_001309831 NP_001309832 NP_001309833 NP_001309834 NP_001309835 NP_001309836 NP_001309837 NP_001309838 NP_001309839 NP_003072 NP_570824 XM_005260808 XM_017028021 XM_017028022 XP_005260865 XP_016883510 XP_016883511</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2 NM_001025081 NM_001025090 NM_001025092 NM_001025094 NM_001025098 NM_001025100 NM_001025101 NM_002385 NP_001020252 NP_001020261 NP_001020263 NP_001020271 NP_001020272 NP_002376 XM_017025778 XM_017025780 XM_024451185 XM_024451186 XM_024451187 XM_024451188 XM_024451189 XP_016881267 XP_016881269 XP_024306953 XP_024306954 XP_024306955 XP_024306956 XP_024306957 XR_001753201 XR_001753202</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3                                                                                                                                                                                                                                                                                                                                                                             NM_020309 NP_064705</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   description      region distance                subregion insideDistance</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1 overlaps 3' overlaps 3'    87299 overlaps exon downstream              0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2 inside exon      inside   153212              inside exon              0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3 inside exon      inside    11788              inside exon              0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   exonnumber nexons            UTR  geneL codingL          Geneid</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1         10     10 overlaps 3'UTR  88588   30705 ENSG00000132639</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2         17     17 overlaps 3'UTR 154856  125322 ENSG00000197971</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3         13     13 overlaps 3'UTR  12959   10860 ENSG00000104888</span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>candidates</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span></span>
<span class='r-in'>    <span class='st'>"chr20:10286777-10288069:+"</span>,</span>
<span class='r-in'>    <span class='st'>"chr18:74690788-74692427:-"</span>,</span>
<span class='r-in'>    <span class='st'>"chr19:49932861-49933829:-"</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'><span class='fu'>region_info</span><span class='op'>(</span><span class='va'>candidates</span>, CSV <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Completed! If CSV=TRUE, check for region_info.csv in the temporary</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> directory (i.e. tempdir()) unless otherwise specified in OUTDIR.</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqnames    start      end width strand    name</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1    chr20 10286777 10288069  1293      +  SNAP25</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2    chr18 74690788 74692427  1640      -     MBP</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3    chr19 49932861 49933829   969      - SLC17A7</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                        annotation</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1                                 NM_001322902 NM_001322903 NM_001322904 NM_001322905 NM_001322906 NM_001322907 NM_001322908 NM_001322909 NM_001322910 NM_003081 NM_130811 NP_001309831 NP_001309832 NP_001309833 NP_001309834 NP_001309835 NP_001309836 NP_001309837 NP_001309838 NP_001309839 NP_003072 NP_570824 XM_005260808 XM_017028021 XM_017028022 XP_005260865 XP_016883510 XP_016883511</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2 NM_001025081 NM_001025090 NM_001025092 NM_001025094 NM_001025098 NM_001025100 NM_001025101 NM_002385 NP_001020252 NP_001020261 NP_001020263 NP_001020271 NP_001020272 NP_002376 XM_017025778 XM_017025780 XM_024451185 XM_024451186 XM_024451187 XM_024451188 XM_024451189 XP_016881267 XP_016881269 XP_024306953 XP_024306954 XP_024306955 XP_024306956 XP_024306957 XR_001753201 XR_001753202</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3                                                                                                                                                                                                                                                                                                                                                                             NM_020309 NP_064705</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   description      region distance                subregion insideDistance</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1 overlaps 3' overlaps 3'    87299 overlaps exon downstream              0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2 inside exon      inside   153212              inside exon              0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3 inside exon      inside    11788              inside exon              0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   exonnumber nexons            UTR  geneL codingL          Geneid</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1         10     10 overlaps 3'UTR  88588   30705 ENSG00000132639</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2         17     17 overlaps 3'UTR 154856  125322 ENSG00000197971</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3         13     13 overlaps 3'UTR  12959   10860 ENSG00000104888</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Sequence</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1                                                                                                                                                                                                                                                                                                                                                            GCTGATTCCAACAAAACCAGAATTGATGAGGCCAACCAACGTGCAACAAAGATGCTGGGAAGTGGTTAAGTGTGCCCACCCGTGTTCTCCTCCAAATGCTGTCGGGCAAGATAGCTCCTTCATGCTTTTCTCATGGTATTATCTAGTAGGTCTGCACACATAACACACATCAGTCCACCCCCATTGTGAATGTTGTCCTGTGTCATCTGTCAGCTTCCCAACAATACTTTGTGTCTTTTGTTCTCTCTTGGTCTCTTTCTTTCCAAAGGTTGTACATAGTGGTCATTTGGTGGCTCTAACTCCTTGAGGTCTTGAGTTTCATTTTTCATTTTCTCTCCTCGGTGGCATTTGCTGAATAACAACAATTTAGGAATGCTCAATGTGCTGTTGATTCTTTCAATCCACAGTATTGTTCTTGTAAAACTGTGACATTCCACAGAGTTACTGCCACGGTCCTTTGAGTGTCAGGCTCTGAATCTCTCAAAATGTGCCGTCTTTGGTTCCTCATGGCTGTTATCTGTCTTTATGATTTCATGATTAGACAATGTGGAATTACATAACAGGCATTGCACTAAAAGTGATGTGATTTATGCATTTATGCATGAGAACTAAATAGATTTTTAGATTCCTACTTAAACAAAAACTTTCCATGACAGTAGCATACTGATGAGACAACACACACACACACAAAACAACAGCAACAACAACAGAACAACAACAAAGCATGCTCAGTATTGAGACACTGTCAAGATTAAGTTATACCAGCAAAAGTGCAGTAGTGTCACTTTTTTCCTGTCAATATATAGAGACTTCTAAATCATAATCATCCTTTTTTAAAAAAAAGAATTTTAAAAAAGATGGATTTGACACACTCACCATTTAATCATTTCCAGCAAAATATATGTTTGGCTGAAATTATGTCAAATGGATGTAATATAGGGTTTGTTTGCTGCTTTTGATGGCTATGTTTTGGAGAGAGCAATCTTGCTGTGAAACAGTGTGGATGTAAATTTTATAAGGCTGACTCTTACTAACCACCATTTCCCCTGTGGTTTGTTATCAGTACAATTCTTTGTTGCTTAATCTAGAGCTATGCACACCAAATTGCTGAGATGTTTAGTAGCTGATAAAGAAACCTTTTAAAAAAATAATATAAATGAATGAAATATAAACTGTGAGATAAATATCATTATAGCATGTAATATTAAATTCCTCCTGTCTCCTCTGTCAGTTTGTGAAGTGATTGACATTTTGTAGCTAGTTTAAAATTATTAAAAATTATAGACTCCAGAT</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 2 GGAGGAAGAGATAGTCGCTCTGGATCACCCATGGCTAGACGCTGAAAACCCACCTGGTTCCGGAATCCTGTCCTCAGCTTCTTAATATAACTGCCTTAAAACTTTAATCCCACTTGCCCCTGTTACCTAATTAGAGCAGATGACCCCTCCCCTAATGCCTGCGGAGTTGTGCACGTAGTAGGGTCAGGCCACGGCAGCCTACCGGCAATTTCCGGCCAACAGTTAAATGAGAACATGAAAACAGAAAACGGTTAAAACTGTCCCTTTCTGTGTGAAGATCACGTTCCTTCCCCCGCAATGTGCCCCCAGACGCACGTGGGTCTTCAGGGGGCCAGGTGCACAGACGTCCCTCCACGTTCACCCCTCCACCCTTGGACTTTCTTTTCGCCGTGGCTGCGGCACCCTTGCGCTTTTGCTGGTCACTGCCATGGAGGCACACAGCTGCAGAGACAGAGAGGACGTGGGCGGCAGAGAGGACTGTTGACATCCAAGCTTCCTTTGTTTTTTTTTCCTGTCCTTCTCTCACCTCCTAAAGTAGACTTCATTTTTCCTAACAGGATTAGACAGTCAAGGAGTGGCTTACTACATGTGGGAGCTTTTGGTATGTGACATGCGGGCTGGGCAGCTGTTAGAGTCCAACGTGGGGCAGCACAGAGAGGGGGCCACCTCCCCAGGCCGTGGCTGCCCACACACCCCAATTAGCTGAATTCGCGTGTGGCAGAGGGAGGAAAAGGAGGCAAACGTGGGCTGGGCAATGGCCTCACATAGGAAACAGGGTCTTCCTGGAGATTTGGTGATGGAGATGTCAAGCAGGTGGCCTCTGGACGTCACCGTTGCCCTGCATGGTGGCCCCAGAGCAGCCTCTATGAACAACCTCGTTTCCAAACCACAGCCCACAGCCGGAGAGTCCAGGAAGACTTGCGCACTCAGAGCAGAAGGGTAGGAGTCCTCTAGACAGCCTCGCAGCCGCGCCAGTCGCCCATAGACACTGGCTGTGACCGGGCGTGCTGGCAGCGGCAGTGCACAGTGGCCAGCACTAACCCTCCCTGAGAAGATAACCGGCTCATTCACTTCCTCCCAGAAGACGCGTGGTAGCGAGTAGGCACAGGCGTGCACCTGCTCCCGAATTACTCACCGAGACACACGGGCTGAGCAGACGGCCCCGTGGATGGAGACAAAGAGCTCTTCTGACCATATCCTTCTTAACACCCGCTGGCATCTCCTTTCGCGCCTCCCTCCCTAACCTACTGACCCACCTTTTGATTTTAGCGCACCTGTGATTGATAGGCCTTCCAAAGAGTCCCACGCTGGCATCACCCTCCCCGAGGACGGAGATGAGGAGTAGTCAGCGTGATGCCAAAACGCGTCTTCTTAATCCAATTCTAATTCTGAATGTTTCGTGTGGGCTTAATACCATGTCTATTAATATATAGCCTCGATGATGAGAGAGTTACAAAGAACAAAACTCCAGACACAAACCTCCAAATTTTTCAGCAGAAGCACTCTGCGTCGCTGAGCTGAGGTCGGCTCTGCGATCCATACGTGGCCGCACCCACACAGCACGTGCTGTGACGATGGCTGAACGGAAAGTGTACACTGTTCCTGAATATTGAAATAAAACAATAAACTTTTAATGGTAT</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                ACACACAGCACATTTCAGCCCCCCAGGCCCCCACCCCCTGTCCGGGACTACTGACCATGTGCCTCCCACTGAATGGCAGTTTCCAGGACCTCCATTCCACTCATCTCTGGCCTGAGTGACAGTGTCAAGGAACCCTGCTCCTCTCTGTCCTGCCTCAGGCCTAAGAAGCACTCTCCCTTGTTCCCAGTGCTGTCAAATCCTCTTTCCTTCCCAATTGCCTCTCAGGGGTAGTGAAGCTGCAGACTGACAGTTTCAAGGATACCCAAATTCCCCTAAAGGTTCCCTCTCCACCCGTTCTGCCTCAGTGGTTTCAAATCTCTCCTTTCAGGGCTTTATTTGAATGGACAGTTCGACCTCTTACTCTCTCTTGTGGTTTTGAGGCACCCACACCCCCCGCTTTCCTTTATCTCCAGGGACTCTCAGGCTAACCTTTGAGATCACTCAGCTCCCATCTCCTTTCAGAAAAATTCAAGGTCCTCCTCTAGAAGTTTCAAATCTCTCCCAACTCTGTTCTGCATCTTCCAGATTGGTTTAACCAATTACTCGTCCCCGCCATTCCAGGGATTGATTCTCACCAGCGTTTCTGATGGAAAATGGCGGTTTCAAGTCCCCGATTCCGTGCCCACTTCACATCTCCCCTACCAGCAGATTCTGCGAAAGCACCAAATTTCTCAAGACCCTCTTCTCCCTAGCTTAGCATAATGTCTGGGGAAACAACCAAAATCGCAATTTTAACAATATGCCTCTCTACCCCCGTGCACTTTTTCTGACATGGTTTTCAGGTCTAAATAGTGGCTGCTCCAGTCCATGAACTCAAAGGTTTGAAGCTACCACCATTGAACTCCCCCATGGTGGTTTCATGATGCCCCCTCCCCAATTCCTCGCACTTTATTCTCCTGGGTGGTTTCGAACTACCCTGTTTCTCAGTGGCCATTTGTTGTGTCCCTCAGGGGCTTAATGACTCAAAAT</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Explore the effect of changing CODING_ONLY</span></span>
<span class='r-in'><span class='co'>## Check how the "distance", "name", "Geneid" among other values change</span></span>
<span class='r-in'><span class='fu'>region_info</span><span class='op'>(</span><span class='st'>"chr10:135379301-135379311:+"</span>, CSV <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Completed! If CSV=TRUE, check for region_info.csv in the temporary</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> directory (i.e. tempdir()) unless otherwise specified in OUTDIR.</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqnames     start       end width strand name annotation description region</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1    chr10 135379301 135379311    11      + &lt;NA&gt;       &lt;NA&gt; inside exon inside</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   distance   subregion insideDistance exonnumber nexons</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1        0 inside exon              0          1      5</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                           UTR geneL codingL          Geneid    Sequence</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1 inside transcription region 63710      NA ENSG00000288107 GCATGTGCGCT</span>
<span class='r-in'><span class='fu'>region_info</span><span class='op'>(</span><span class='st'>"chr10:135379301-135379311:+"</span>, CSV <span class='op'>=</span> <span class='cn'>FALSE</span>, CODING_ONLY <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Completed! If CSV=TRUE, check for region_info.csv in the temporary</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> directory (i.e. tempdir()) unless otherwise specified in OUTDIR.</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqnames     start       end width strand   name          annotation</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1    chr10 135379301 135379311    11      + CYP2E1 NM_000773 NP_000764</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   description     region distance subregion insideDistance exonnumber nexons</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1  downstream downstream    45391      &lt;NA&gt;             NA         NA     12</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    UTR geneL codingL          Geneid    Sequence</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> 1 &lt;NA&gt; 40814   11568 ENSG00000130649 GCATGTGCGCT</span>
<span class='r-in'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'><span class='fu'>region_info</span><span class='op'>(</span><span class='va'>candidates</span>, OUTDIR <span class='op'>=</span> <span class='st'>"/path/to/directory/"</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='fu'>region_info</span><span class='op'>(</span><span class='st'>"chr20:10286777-10288069:+"</span>, OUTDIR <span class='op'>=</span> <span class='st'>"/path/to/directory"</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>}</span></span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Amanda Price.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


